### ISSUE TITLE
Rewrite the book to use Haskell instead of Scala

### ISSUE THREAD

**Issue by winitzki** [winitzki/sofp#3]
- [ ] Go through all code examples and try to write them in Haskell instead of Scala. This will discover which code examples have no reasonable Haskell equivalent at the same pedagogical level as appropriate at each place in the text.
- [ ] Go through all exercises and try to implement them in Haskell. See if there are any problems specific to the differences between Haskell and Scala.
- [ ] Identity all text that specifically addresses features of Scala that do not have any counterparts in Haskell. See if that text can be removed without disturbing the flow of the presentation, or can be replaced by some Haskell-specific equivalent.

**Comment by leobenkel** [comment]
is it a second version of the same book in Haskell or will that replace the scala version ?

**Comment by winitzki** [comment]
No, I'm not going to replace the Scala version, because there is some Scala-specific material that is not applicable to Haskell, and vice versa. Also, because Scala is a lot more popular, and there is a larger audience for a Scala version of this book. But a Haskell version would be a good thing as well.
